### PR TITLE
Fixed a velocypack type error in maintenance::reportInCurrent

### PR DESCRIPTION
**Pull request by goedderz**
### Scope & Purpose

In a velocypack type change introduced in #12594, an error-handling branch was overlooked, rendering the error-handling code useless, always resulting in an exception (or assertion in maintainer mode).

Fixes errors like the following:

```
WARNING [ceb1a] {maintenance} caught exception in Maintenance shards error reporting: Expecting Object
ERROR [c9a75] {maintenance} Error reporting in current: Expecting Object
```

It seems there was a previous effort to find the source of this error, or a similar one, in #13342.

- [X] :hankey: Bugfix (requires CHANGELOG entry)
- [ ] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification
- [x] :book: CHANGELOG entry made

#### Backports:

As far as I can tell, the PR introducing the bug was merged into >=3.7, but not into 3.6.

- [X] Backport required for 3.8: #14472
- [X] Backport required for 3.7: #14473

#### Related Information

*(Please reference tickets / specification / other PRs etc)* **TODO**

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket number:
- [ ] Design document: 

### Testing & Verification

**TODO** needs additional unit tests. This code is currently completely untested!

- [ ] This change is a trivial rework / code cleanup without any test coverage.
- [ ] The behavior in this PR was *manually tested*
- [ ] This change is already covered by existing tests, such as *(please describe tests)*.
- [ ] This PR adds tests that were used to verify all changes:
  - [ ] Added new C++ **Unit tests**
  - [ ] Added new **integration tests** (e.g. in shell_server / shell_server_aql)
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)
- [ ] There are tests in an external testing repository:
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

### CODE REVIEW

File: arangod/Cluster/Maintenance.cpp [hunk 1]
```diff
@@ -1466,7 +1466,9 @@ arangodb::Result arangodb::maintenance::reportInCurrent(
 
           std::string c = key.substr(pos + 1, pos2);
           std::string s = key.substr(pos2 + 1);  // shard name
-          auto const pdb = pit->second->slice();
+          TRI_ASSERT(pit->second->slice().isArray());
+          TRI_ASSERT(pit->second->slice().length() == 1);
```
Comment (gnusi) [inline arangod/Cluster/Maintenance.cpp:1470]: how can we guarantee that?

Comment (goedderz) [reply arangod/Cluster/Maintenance.cpp:1470]: Only by the way the `plan` argument is created in the first place. Same holds for `current` and `local`. The proper solution IMO would be to replace VelocyPack with static data structures. But that touches the whole maintenance.

So we should introduce tests in this PR for `reportInCurrent`, and the code that writes it input in the first place.

Comment (kvahed) [reply arangod/Cluster/Maintenance.cpp:1470]: @gnusi, yes it is guaranteed. You will not ever get here without a plan object.

**Review by kvahed — APPROVED** [review]
LGTM

File: arangod/Cluster/Maintenance.cpp (hunk 1 above)
Comment (gnusi) [reply arangod/Cluster/Maintenance.cpp:1470]: @kvahed having an array of exactly one element looks really strange. Where is the code building this piece of the plan located?

**Pull request merged**